unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -209,6 +308,10 @@ public boolean equals(final Object other) {
 
   @Override
   public int hashCode() {
-    return Arrays.hashCode(bytes);
```

Comment:
If performance is important here, we could consider adding a `hashUnsafeBytes` method to `Murmur3_x86_32` and could use that instead.  We can also defer that to a followup PR, since I'd like to get this merged soon.